### ISSUE TITLE
Explicitly typed variables in base class are shared across derived classes instead of separate instances

### ISSUE THREAD

**Issue by epark009** [godotengine/godot#33100]
**Godot version:**
v.3.1.1.stable.official

**OS/device including version:**
Windows 10
Intel(R) Core(TM) i7-7700 CPU @ 3.60GHz, 3601 Mhz, 4 Core(s), 8 Logical Processor(s)
Nvidia GeForce GTX 1080 Ti

**Issue description:**
When there's a base class with an explicitly typed variable (i.e. var a: Array), inherit this base class into a different script, then make multiple instances of the scene containing that script. That variable gets shared across all instances of the scene instead of separate as expected. If the variable is implicitly typed (i.e. var a = []) then this works as expected.

**Steps to reproduce:**
Create a base class with an array or dictionary:

```
extends Node2D
class_name BaseClass

var a: Array
#var a = []

func _ready():
	pass

func do(x):
	a.append(x)
	print(a)
```

Inherit the base class in a script for some node in some scene:

```
extends BaseClass

func _ready():
	pass
```

Let's call this scene Node2D. In another scene, Main, create at least 2 instances of the Node2D scene. Then in Main, create a script that calls the inherited `do` function:

```
extends Node2D

onready var node1 = $"Node2D"
onready var node2 = $"Node2D2"

# Called when the node enters the scene tree for the first time.
func _ready():
	pass # Replace with function body.

func _process(delta):
	node1.do(1)
	node2.do(2)
```

Expected output:
```
[1]
[2]
[1, 1]
[2, 2]
[1, 1, 1]
[2, 2, 2]
etc.
```

Actual output:
```
[1]
[1, 2]
[1, 2, 1]
[1, 2, 1, 2]
[1, 2, 1, 2, 1]
[1, 2, 1, 2, 1, 2]
etc.
```

In the base class, if you replace `var a: Array` with `var a = []`, then the actual output matches the expected output.

**Minimal reproduction project:**
[test.zip](https://github.com/godotengine/godot/files/3775428/test.zip)

**Comment by akien-mga** [comment]
Duplicate of #23134

**Comment by KoBeWi** [comment]
More like duplicate of #32941, but still irrelevant (fixed in 3.2).
